packages/create-pantheon-decoupled-kit/src/templates/partials/next-wp/Posts.ts

```diff
@@ -0,0 +1,167 @@
+import { taggedTemplateHelpers as utils } from '@cli/utils';
+
+const postUriAcf = /* js */ `
+export async function getPostByUri(uri) {
+	const uriString = uri.join('/');
+	const query = gql ${utils.backticks`
+		query PostBySlugQuery($uriString: ID!) {
+			post(id: $uriString, idType: URI) {
+				title
+				date
+				featuredImage {
+					node {
+						altText
+						sourceUrl
+					}
+				}
+				content
+				relatedContent {
+					relatedPosts {
+						... on Post {
+							id
+							uri
+							title
+							featuredImage {
+								node {
+									altText
+									sourceUrl
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`};
+
+	const {
+		data: { post },
+		headers,
+	} = await client.rawRequest(query, { uriString });
+
+	return { post, headers };
+}
+`;
+
+const postUri = /* js */ `
+export async function getPostByUri(uri) {
+	const uriString = uri.join();
+	const query = gql ${utils.backticks`
+		query PostBySlugQuery($uriString: ID!) {
+			post(id: $uriString, idType: URI) {
+				title
+				date
+				featuredImage {
+					node {
+						altText
+						sourceUrl
+					}
+				}
+				content
+			}
+		}
+	`};
+
+	const {
+		data: { post },
+		headers,
+	} = await client.rawRequest(query, { uriString });
+
+	return { post, headers };
+}
+`;
+const searchQuery = /* js */ `
+export async function getSearchedPosts(searchTerm) {
+	const query = gql ${utils.backticks`
+		query LatestPostsQuery($searchTerm: String) {
+			posts(where: { search: $searchTerm }) {
```

Comment:
@mitchellmarkoff @CobyPear this query looks pretty much identical to what the query would be if you weren't using Solr. It is also referencing posts rather than an index. Did you confirm that this is actually taking advantage of the Solr index you configured? A couple of ways you could do that:

* Assuming that the index you created indexes both posts and pages (the acceptance criteria in the ticket specifies that this search should return posts and pages) try searching for something that should match pages but not posts. If you get no results, this might not be using the index.
* Alternatively, point this to a WP instance that doesn't have Solr configured. if you still get results, that indicates that this isn't using Solr at all.

My best guess is that this is using default WordPress search, but maybe the Solr plugin does some magic I'm not aware of (the docs do say things like 'Completely replaces default WordPress search'.) Didn't mean to lead you in the wrong direction when asking if we could get at Solr data via GraphQL.

Reply:
Oh good call. I'm not seeing any Pages come back from the search.

Reply:
@backlineint  Here is what I had done to test that Solr is being used here, I used two CMS sites. One without Solr enabled and one with Solr.

Each of these CMS sites have 3 different posts titled `A Example`, `B Example`, and `C Example`. These posts were created in the order I just listed them.

Expectations when querying the term `example`:
- For the CMS without Solr, I would expect to get results from the default WP search that are ordered by the date/time they were created, with the newest post being shown first. Expected order - `C Example`, `B Example`, `A Example`.
- For the CMS with Solr, I would expect the results to come back based on score, which in this case should place a higher emphasis on alphabetical order since the search term is present in the same place of each of these posts. Expected order - `A Example`, `B Example`, and `C Example`

When I went to confirm this I got results that matched the expectations above and lead me to believe that Solr is working as expected here.

Feel free to checkout the test sites I used:
Without Solr: https://dev-gql-search.pantheonsite.io/wp-admin/admin.php?page=graphiql-ide
With Solr: https://dev-wp-search-demo.pantheonsite.io/wp-admin/admin.php?page=graphiql-ide
Normal WP QA credentials for both.

Query used in each:
```
query NewQuery {
  posts(where: {search: "example"}) {
    nodes {
      title
      date
    }
  }
}
```

Results from site without Solr:
```
{
  "data": {
    "posts": {
      "nodes": [
        {
          "title": "C Example",
          "date": "2023-08-22T15:40:42"
        },
        {
          "title": "B Example",
          "date": "2023-08-22T15:39:53"
        },
        {
          "title": "A Example",
          "date": "2023-08-22T15:39:37"
        }
      ]
    }
  },
```

Results from site with Solr:
```
{
  "data": {
    "posts": {
      "nodes": [
        {
          "title": "A Example",
          "date": "2023-08-22T15:41:23"
        },
        {
          "title": "B Example",
          "date": "2023-08-22T15:41:34"
        },
        {
          "title": "C Example",
          "date": "2023-08-22T15:41:43"
        }
      ]
    }
  },
```